cmd/simulator/load/workers.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"context"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/sync/errgroup"
+)
+
+type WorkerGroup struct {
+	Workers []*Worker
+}
+
+func NewWorkerGroup(clients []ethclient.Client, senders []common.Address, txSequences []types.Transactions) *WorkerGroup {
+	workers := make([]*Worker, len(clients))
+	for i, client := range clients {
+		workers[i] = NewWorker(client, senders[i], txSequences[i])
+	}
+
+	return &WorkerGroup{
+		Workers: workers,
+	}
+}
+
+func (wg *WorkerGroup) executeTask(ctx context.Context, f func(ctx context.Context, w *Worker) error) error {
+	eg := errgroup.Group{}
+	for _, worker := range wg.Workers {
+		worker := worker
+		eg.Go(func() error {
+			return f(ctx, worker)
+		})
+	}
+
+	return eg.Wait()
+}
+
```

Comment:
do these functions need to be public?

Reply:
Yes, these are intended to be public so that an external package can execute them step by step.